### PR TITLE
refactored unreadable or inefficient code

### PR DESCRIPTION
**Pull request by jasperdewinther**
Refactored dictionary to be efficient, moved big functions into smaller readable functions. And made the load command usable for other locations than root.

### CODE REVIEW

File: module/frame_parser.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import re
```
Comment (Timmotus) [inline module/frame_parser.py]: this file should be removed as it has [already been implemented in python build](https://github.com/R2D2-2019/r2d2-python-build/pull/31)

File: module/cli_controller.py [hunk 2]
```diff
@@ -120,49 +127,71 @@ def ask_input(input_queue:queue.Queue, string=""):
...
+    Go back one node in the tree structure. You cant go back when in root
+    """
+    def go_back_in_tree(self):
+        if self.current_node.parent:
+            self.current_node = self.current_node.parent
+
+    """
+    Go to root in tree structure
+    """
+    def goto_root(self):
```
Comment (Timmotus) [inline module/cli_controller.py]: go_to_root instead of goto_root

File: module/cli_controller.py [hunk 3]
```diff
@@ -100,14 +107,14 @@ def make_path_string(path_list):
     """
     Prints the info and any children or parameters of the node.
     """
-    @staticmethod
-    def print_help(node):
+    def print_help(self):
```
Comment (Timmotus) [inline module/cli_controller.py:110]: what is the reason this did not stay static?

File: module/cli_controller.py [hunk 4]
```diff
@@ -37,12 +42,12 @@ def __init__(self, comm:BaseComm, command_file_list=None):
...
     def load_tree(self):
         for file in self.command_file_list:
-            self.load_commands(file)
+            self.load_commands(file, self.root_node)
 
     """
     Loads a single JSON file into command structure
     """
-    def load_commands(self, file):
+    def load_commands(self, file, rootnode):
```
Comment (Timmotus) [inline module/cli_controller.py]: rootnode does not comply with the style guide, should be either root_node or a different name

**Review by Timmotus — CHANGES_REQUESTED** [review]

File: module/cli_controller.py (hunk 3 above)
Comment (jasperdewinther) [reply module/cli_controller.py:110]: This uses self now. This was changed so now the function does not require any arguments and that makes it a lot simpler in the global_commands dict.

**Review by Timmotus — APPROVED** [review]

**Pull request merged**